src/server/controllers/user.js

```diff
@@ -3,37 +3,37 @@ const jwt = require('jsonwebtoken');
 const { PrismaClient } = require('@prisma/client');
 const prisma = new PrismaClient();
 
-const jwtSecret = 'mysecret';
+const key = '87764d1a-92dc-4ced-a758-9c898c31d525'
+const saltRounds = 10;
+const maxAge = 7 * 24 * 60 * 60
 
 const register = async (req, res) => {
-    const { username, password } = req.body;
-
-    const createdUser = null;
-
-    res.json({ data: createdUser });
+  const { username, password } = req.body;
+  const hash = await bcrypt.hash(password, saltRounds)
+  const createdUser = await prisma.user.create({
+    data: { username, password: hash }
+  })
+  res.json({ data: createdUser });
 };
 
 const login = async (req, res) => {
-    const { username, password } = req.body;
-
-    const foundUser = null;
-
-    if (!foundUser) {
-        return res.status(401).json({ error: 'Invalid username or password.' });
-    }
-
-    const passwordsMatch = false;
-
-    if (!passwordsMatch) {
-        return res.status(401).json({ error: 'Invalid username or password.' });
-    }
-
-    const token = null;
-
-    res.json({ data: token });
+  const { username, password } = req.body;
+  const foundUser = await prisma.user.findUnique({
+    where: { username }
+  });
+
+  if (!foundUser) {
+    return res.status(401).json({ error: 'Invalid username or password.' });
+  }
+  const match = await bcrypt.compare(password, foundUser.password)
+  if (!match) {
+    return res.status(401).json({ error: 'Invalid username or password.' });
+  }
+  const token = jwt.sign({ userId: foundUser.id }, key, { expiresIn: maxAge });
```

Comment:
Added an expiry to the token, fantastic. I guess this puts a timer on being able to access it in storage?

Reply:
exactly! I tried to do it with cookies (not recommended), but this is way easier